### DISCUSSION TITLE
Hover variant overrides a custom hover utility class styles in plugin

### DISCUSSION THREAD

**Discussion by mightymatth** [tailwindlabs/tailwindcss#11153] [Help] [answered]
**What version of Tailwind CSS are you using?**

v3.3.2

**What build tool (or framework if it abstracts the build tool) are you using?**

Tailwind Playground

**What version of Node.js are you using?**

Tailwind Playground

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/3lvxwrvQBx

**Describe your issue**

I have a custom CSS styling for `hover` in a 3rd-party library in _CSS-in-JS_ syntax that I import as a utility class in a custom Tailwind plugin. 

The whole setup looks like this:
```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  plugins: [
    plugin(({ addUtilities }) => {
      addUtilities({
        '.custom-hover': getHoverStyle,
      });
    })
  ],
}

// style from a 3rd-party library
const getHoverStyle = {
    borderRadius: "medium",
    marginLeft: "-2px",
    marginRight: "-2px",
    paddingLeft: "2px",
    paddingRight: "2px",
    '@media(hover:hover)': {
        '&': {
            transition: "background var(--p-transition-duration, .24s) ease",
        },
        '&:hover': {
            background: "rgba(148, 149, 152, .18)",
        },
    },
};

```

In template, I have this:
```html
<span class="custom-hover">custom-hover (works)</span>

<span class="hover:custom-hover">hover:custom-hover (doesn't work)</span>
```

The problem is that, if I want to use `hover:custom-hover`, it wouldn't work because it gets overwritten by `hover` variant, I suppose. (denied in EDIT1).

1. Is there are way to fix this issue just by proper configuration?
  - or modifying `getHoverStyle` CSS-in-JS variable is the only way?
2. Do you consider this as a correct behavior or maybe a bug?
  - if it's clear, move to discussions/help, please.

Thanks!

---

EDIT1:

I noticed that it generated a broken CSS when using it with the hover variant:

<details>
<summary>Generated CSS - Utilities</summary>

```
.custom-hover {
  border-radius: medium;
  margin-left: -2px;
  margin-right: -2px;
  padding-left: 2px;
  padding-right: 2px;
  @media(hover:hover) {
    & {
      transition: background var(--p-transition-duration, .24s) ease;
    }

    &:hover {
      background: rgba(148, 149, 152, .18);
    }
  }
}

.hover\:custom-hover:hover {
  border-radius: medium;
  margin-left: -2px;
  margin-right: -2px;
  padding-left: 2px;
  padding-right: 2px;
  @media(hover:hover) {
     {
      transition: background var(--p-transition-duration, .24s) ease;
    }

     {
      background: rgba(148, 149, 152, .18);
    }
  }
}
```
</details>

**Comment by adamwathan** [comment, marked as answer]
Hey! If you add a space after `@media` it generates the CSS I'd expect:

https://play.tailwindcss.com/3ntzAr2uf9?file=config

It's still kinda nonsense to me though, you have a utility that already has hover styles built-in but then you want to only enable it on hover which leads to stuff like `.hover\:custom-hover:hover:hover` where the `:hover` pseudo-class appears twice.

If you can describe the effect you're trying to achieve here there's probably a more idiomatic way to handle this with Tailwind than what you're doing I think — going to move this to a discussion but happy to help you figure it out!

**Reply by mightymatth** [reply]
Oh, thanks @adamwathan, I didn't have a clue that it could be the issue. 😓 

> It's still kinda nonsense to me though, you have a utility that already has hover styles built-in but then you want to only enable it on hover

Completely agree. The naming is bad and there is no sense to do it, I just noticed it and couldn't figure out why.

To give you some background, I'm trying to create a preset that defines the specific defaults of the design system that we use. We have this const `getHoverStyle` that is, in my opinion, badly named because it covers both normal and hover states. `getLinkStyle` might be better, but I'm not sure atm.

However, don't you think that this missing space would potentially cause issues for somebody else? As I can [see](https://github.com/search?q=%2F%28%22%7C%27%29%40media%5C%28%2F+path%3A%2F.*%5C.%28js%7Cjsx%7Cts%7Ctsx%29%24%2F&type=code), there is a lot of code written like that. If it's not on the Tailwind side, we could raise the issue somewhere else.

Thanks for debugging :)
